DiscordRichPresence/DiscordRichPresence.cpp

```diff
@@ -93,38 +93,45 @@ int init()
 
     LoadSettingsFile();
 	
-	g_presenceInfo.InitializeDiscordRPC();
+	if (g_pluginSettings.SendWhenPausedOrStopped)
+	{
+		g_presenceInfo.InitializeDiscordRPC();
 
-    ReportIdleStatus();
+    	ReportIdleStatus();
+	}
 
     return 0;
 }
 
 void ReportIdleStatus()
-{
-    if (!g_presenceInfo.HasDiscordModuleLoaded())
+{    
+	if (g_pluginSettings.ApplicationID == "0")
         return;
 
-    if (g_pluginSettings.ApplicationID == "0")
-        return;
+	if (!g_presenceInfo.IsDiscordRPCConnected())
+	{
+        g_presenceInfo.InitializeDiscordRPC();
+	}
 
 	g_presenceInfo.CurrentPlaybackState = Stopped;
 	g_presenceInfo.SetStartTimestamp(0);
 	g_presenceInfo.SetStateText("(Idle)");
 	g_presenceInfo.ClearDetails();
-
-	g_presenceInfo.PostToDiscord();
+	g_presenceInfo.PostToDiscord();	
 }
 
 void ReportCurrentSongStatus(PlaybackState playbackState)
 {
-    if (!g_presenceInfo.HasDiscordModuleLoaded())
-        return;
+	assert(playbackState != Stopped);
 
     if (g_pluginSettings.ApplicationID == "0")
```

Comment:
Comment about pre-existing- not a requested action
Been meaning to change this. I hate unnecessary string compares

Reply:
Are you talking about the ApplicationID check?

I'd like to expose a 'validate' function or similar that gate the connection... or even whisk all of the 'get connected' or 'get disconnected' into a wrapper function to encapsulate that logic.

Reply:
yeah, the ApplicationID check. For some reason it highlighted the wrong line. Not trying to de-rail your PR, this is probably another thing for another day.